### PR TITLE
Validate k8s pods resource

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Resources would be checked when using pod tasks like they are checked in python tasks.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Invoke `validateK8sPod` if running pod tasks

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2187

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/task_validator.go [hunk 1]
```diff
@@ -44,6 +42,31 @@ func validateContainer(task core.TaskTemplate, taskConfig runtime.TaskResourceCo
 	return nil
 }
 
+// This is called for a task with a non-nil k8s pod.
+func validateK8sPod(task core.TaskTemplate, taskConfig runtime.TaskResourceConfiguration) error {
+	if task.GetK8SPod().PodSpec == nil {
+		return nil
```
Comment (katrogan) [inline pkg/manager/impl/validation/task_validator.go:48]: is this actually valid?

**Review by katrogan — DISMISSED** [review]

File: pkg/manager/impl/validation/task_validator.go (hunk 1 above)
Comment (pingsutw) [reply pkg/manager/impl/validation/task_validator.go]: Updated it, thanks

File: pkg/repositories/database_integration_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build integration
```
Comment (katrogan) [inline pkg/repositories/database_integration_test.go:1]: why add this?

Comment (pingsutw) [reply pkg/repositories/database_integration_test.go:1]: This is automatically generated by `goimports` (goimports -w pkg/*) 
Could we add this? So we don't need to manually delete it when running `goimports`

Comment (pingsutw) [reply pkg/repositories/database_integration_test.go:1]: thank you

**Pull request merged**